src/transformers/models/propainter/image_processing_propainter.py

```diff
@@ -0,0 +1,540 @@
+                    crop_size=crop_size,
+                    do_rescale=do_rescale,
+                    rescale_factor=rescale_factor,
+                    do_normalize=do_normalize,
+                    image_mean=image_mean,
+                    image_std=image_std,
+                    data_format=data_format,
+                    input_data_format=input_data_format,
+                    is_mask_frame = False,
+                )
+                for img in video
+            ]
+            for video in videos
+        ]
+
+        pixel_values_masks = []
+        if scale_hw is None:
+            for mask in masks:
+                pixel_values_mask = [
+                        self._preprocess_image(
+                            image=img,
+                            do_resize=True,
+                            size={"height": video_size[0], "width": video_size[1]},
+                            resample=PILImageResampling.NEAREST,
+                            do_center_crop=do_center_crop,
+                            crop_size=crop_size,
+                            do_rescale=True,
+                            rescale_factor=255,
+                            do_normalize=do_normalize,
+                            image_mean=image_mean,
+                            image_std=image_std,
+                            data_format=data_format,
+                            input_data_format=input_data_format,
+                            is_mask_frame = True,
+                            mask_dilation= mask_dilation,
+                        )
+                        for img in mask
+                    ]
+                if len(mask) == 1:
+                  pixel_values_mask = pixel_values_mask * len(videos[0])
+                pixel_values_masks.append(pixel_values_mask)
+        else:
+            #for outpainting of videos
+            videos, flow_masks, masks_dilated, (height_extr, width_extr)= [extrapolation(
+                        video, scale = scale_hw, num_frames=len(video))
+                        for video in videos]
+
+        if scale_hw is None:
+            #masks is for both flow_masks, masks_dilated, just add the same data to both variables in case of inpainting
+            flow_masks = masks_dilated = torch.stack([to_tensors(mask) for mask in pixel_values_masks], dim=0)
+            size = video_size #height, width
+        else:
+            flow_masks, = torch.stack([to_tensors(mask) for mask in flow_masks], dim=0)
+            masks_dilated = torch.stack([to_tensors(mask) for mask in masks_dilated], dim=0)
+            size = (height_extr, width_extr)
+        videos_inp = [[np.array(frame).transpose(1,2,0).astype(np.uint8) for frame in video] for video in videos][0]
+        videos = torch.stack([to_tensors(video) * 2 - 1 for video in videos], dim=0)
+
+        data = {"pixel_values_inp": videos_inp, "pixel_values": videos, "flow_masks": flow_masks, "masks_dilated": masks_dilated, "size": size}
```

Comment:
This input name has to be changed. It is basically a numpy array of original frames (before they are converted to tensors). Well, it's just extra code and this can just be done in modeling file from `pixel_values` input. So, this will be removed in the next commits

Reply:
Update: removed